saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/registration/OpenSamlMetadataAssertingPartyDetailsConverter.java

```diff
@@ -216,4 +223,65 @@ private Document document(InputStream inputStream) {
 		return new ArrayList<>();
 	}
 
+	public static final class AssertingPartyDetails extends RelyingPartyRegistration.AssertingPartyDetails {
```

Comment:
I am wondering why you have decided to have an inner class instead `OpenSamlAssertingPartyDetails`?